File: Java/src/p_3_creational_patterns/p_9_builder/SaleManager.java
package p_3_creational_patterns.p_9_builder;

public class SaleManager {
    private OrderBuilder _orderBuilder;

    public Car createOrder( String brandName, String modelName, String colorName, int power){
        if ( brandName.equals("AUDI"))
            _orderBuilder = new AudiOrderBuilder();
        else if ( brandName.equals("FIAT"))
            _orderBuilder = new FiatOrderBuilder();

        _orderBuilder.setBrand(brandName);
        _orderBuilder.setModel(modelName);
        _orderBuilder.setColor(colorName);
        _orderBuilder.setPower(power);

        return _orderBuilder.getOrderedCar();
    }

    public void printOrder()
    {
        System.out.println(_orderBuilder.getOrderedCar().getBrand());
        System.out.println(_orderBuilder.getOrderedCar().getModel());
        System.out.println(_orderBuilder.getOrderedCar().getColor());
        System.out.println(_orderBuilder.getOrderedCar().getPower());
    }
}

File: Net/03_creational_patterns/05_builder_pattern/SaleManager.cs
namespace _05_builder_pattern
{
    public class SaleManager
    {
        private OrderBuilder _orderBuilder;

        public Car createOrder(String brandName, String modelName, String colorName, int power)
        {
            if (brandName.Equals("AUDI"))
                _orderBuilder = new AudiOrderBuilder();
            else if (brandName.Equals("FIAT"))
                _orderBuilder = new FiatOrderBuilder();

            _orderBuilder.SetBrand(brandName);
            _orderBuilder.SetModel(modelName);
            _orderBuilder.SetColor(colorName);
            _orderBuilder.SetPower(power);

            return _orderBuilder.GetOrderedCar();
        }

        public void printOrder()
        {
            Console.WriteLine(_orderBuilder.GetOrderedCar().GetBrand());
            Console.WriteLine(_orderBuilder.GetOrderedCar().GetModel());
            Console.WriteLine(_orderBuilder.GetOrderedCar().GetColor());
            Console.WriteLine(_orderBuilder.GetOrderedCar().GetPower());
        }
    }
}

